File: workspaces/jflobos/linked-list-cycle.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} head
 * @return {boolean}
 */
var hasCycle = function(head) {
    if (head === null) {
        return false;
    }
    let turtleHead = head.next;
    let rabbitHead;
    if (head.next !== null) {
        rabbitHead = head.next.next;
    } else {
        rabbitHead = null;
    }

    while (turtleHead !== null && rabbitHead !== null) {
        if (turtleHead === rabbitHead) {
            return true;
        }
        turtleHead = turtleHead.next;
        if (rabbitHead.next === null) {
            rabbitHead = null;
        } else {
            rabbitHead = rabbitHead.next.next;
        }
    }
    return false;
};

File: workspaces/fcopardo/app/src/main/kotlin/fcopardo/linkedList/LinkedListCycle.java
package app.src.main.kotlin.fcopardo.linkedList;

public class LinkedListCycle {
    public boolean hasCycle(ListNode head) {

        ListNode one = head;
        ListNode two = head;
        while(two != null && two.next !=null){
            one = one.next;
            two = two.next.next;
            if(two == one) return true;
        }
        return false;
    }
}
